Mueez Afzal-weatherman.py

```diff
@@ -0,0 +1,304 @@
+            month_count+=1
+    return year_list,month_list,day_list                
+
+def weather_report(year_list):
+    stop="y"
+    input_list=[]
+    while stop=="y" or stop=="Y":
+
+         while True:
+             try:
+                year_name=int(input("Enter the Year from 2004 to 2016 :"))
+                if year_name < 2004 or year_name > 2016:
+                      raise ValueError #this will send it to the print message and back to the input option
+                input_list.append(year_name)
+                break
+             except ValueError:
+                     print("Invalid integer. The number must be in the range of 2004-2016.")
+             
+         stop=input("Want to enter more y/n :")
+
+    for name in range(len(year_list)):
+        for input_string in input_list:
+           if str(input_string) == year_list[name].get_name():
+               year_list[name].print_name()
+               
+
+             
+def year_report(year_list):
+  
+    year_name=year_input()    
+    for name in range(len(year_list)):
+        if str(year_name) == year_list[name].get_name():
+            year_list[name].print_name()
+            break
+
+def month_report(month_list):
+
+    year_name=year_input()
+    month_num=month_input()
+ 
+    month_name=months.get(str(month_num))
+    for name in range(len(month_list)):
+        if str(year_name) == month_list[name].get_year() and str(month_name) == month_list[name].get_month():
+            month_list[name].print_name()
+            break
+        if name==(len(month_list)-1):
+            print("No record")
+def day_report(day_list):
+    year_name=year_input()
+    month_name=month_input()
+    month_name=months.get(str(month_name))
+    for name in range(len(day_list)):
+        if str(year_name) == day_list[name].get_year() and str(month_name) == day_list[name].get_month():
+            day_list[name].print_name()           
+
+def year_input():
+    while True:
+       try:
+           year_name=int(input("Enter the Year 2004 to 2016:"))
+           if  year_name< 2004 or year_name> 2016:
+              raise ValueError #this will send it to the print message and back to the input option
+           break
+       except ValueError:
+              print("Invalid integer. The number must be in the range of 2004-2016.")
+    return year_name
+
+def month_input():
+   while True:
+       try:
+              month_name=int(input("Enter the month 1_12:"))
+              if  month_name <1  or  month_name > 12:
+                   raise ValueError #this will send it to the print message and back to the input option
+              break
+       except ValueError:
+              print("Invalid integer. The number must be in the range of 1 to 12.")
+   return month_name
+
+def main():
+    day_list=[]
+    month_list=[]
+    year_list=[]
+    year_list,month_list,day_list=file_data(day_list,month_list,year_list)
+    while True:

```

Comment:
We don't need infinite loops in this program.
the program should take command line arguments. 
Please read the requirements carefully.